plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/recipes/CreateDynamicJenkinsFile.java

```diff
@@ -0,0 +1,147 @@
+package io.jenkins.tools.pluginmodernizer.core.recipes;
+
+import io.jenkins.tools.pluginmodernizer.core.extractor.ArchetypeCommonFileVisitor;
+import io.jenkins.tools.pluginmodernizer.core.extractor.PluginMetadata;
+import io.jenkins.tools.pluginmodernizer.core.extractor.PomResolutionVisitor;
+import io.jenkins.tools.pluginmodernizer.core.model.JDK;
+import io.jenkins.tools.pluginmodernizer.core.utils.JsonUtils;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import org.openrewrite.*;
+import org.openrewrite.text.CreateTextFile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CreateDynamicJenkinsFile extends ScanningRecipe<CreateDynamicJenkinsFile.ConfigState> {
+    private static final Logger LOG = LoggerFactory.getLogger(CreateDynamicJenkinsFile.class);
+
+    @Option(displayName = "Version", description = "The version.", example = "2.452.4")
```

Comment:
This recipe must not have parameters since the Jenkins version is taken from the pom.xml